### DISCUSSION TITLE
Odd panning behavior in 3D panel

### DISCUSSION THREAD

**Discussion by yepterence** [foxglove/community#970] [Bugs] [answered]
**Description**
When we click and drag to pan across in the 3D panel, we notice that the cursor tends not to return to the same spot that you've clicked and held during panning.
This issue wan't present when we visualized our ROS1 bags with webviz.

- Version: 1.84.0
- Platform: Foxglove-Studio
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native): ROS1 bag/ROS1 mcap

**Steps To Reproduce**

Expected behavior video
[webviz-expected-behavior.webm](https://github.com/foxglove/studio/assets/23585808/43dc6190-8e69-4cfb-87b5-50db108cca9c)

Unexpected Behavior video
[foxglove-studio-unexpected-behavior.webm](https://github.com/foxglove/studio/assets/23585808/d5af1d2a-a890-48d8-9500-d4b9db539e4b)

**Expected Behavior**
After you're done panning, I would expect that the mouse cursor is on top of the last thing it was clicked and dragged on, if the mouse button is clicked and held.

**Comment by foxhubber** [comment]
Internal tracking ticket: [FG-6537](https://linear.app/foxglove/issue/FG-6537/odd-panning-behavior-in-3d-panel-[foxglovestudio-7458])

**Comment by amacneil** [comment]
Thanks for reporting, I can reproduce (below) and we're taking a look at it.

My description of the issue (assuming I understand you correctly): When dragging in 2D mode, the cursor is moving faster than the underlying canvas. Instead the cursor should stay "attached" to the canvas while dragging.

https://github.com/foxglove/studio/assets/637671/65927f85-dad9-443d-ad40-00003400de33

**Comment by amacneil** [comment, marked as answer]
Update: This is fixed in 2.0
